hw/sensor/src/sensor.c

```diff
@@ -1162,18 +1184,26 @@ sensor_interrupt_ev_cb(struct os_event *ev)
 static void
 sensor_notify_ev_cb(struct os_event * ev)
 {
-    const struct sensor_notify_ev_ctx *ctx;
+    struct sensor_notify_os_ev *snoe;
     const struct sensor_notifier *notifier;
 
-    ctx = ev->ev_arg;
+    snoe = ev->ev_arg;
+
+    if (!snoe) {
+        return;
+    }
 
-    SLIST_FOREACH(notifier, &ctx->snec_sensor->s_notifier_list, sn_next) {
-        if (notifier->sn_sensor_event_type & ctx->snec_evtype) {
-            notifier->sn_func(ctx->snec_sensor,
+    SLIST_FOREACH(notifier, &snoe->snoe_sensor->s_notifier_list, sn_next) {
+        if (notifier->sn_sensor_event_type & snoe->snoe_evtype) {
+            notifier->sn_func(snoe->snoe_sensor,
                               notifier->sn_arg,
-                              ctx->snec_evtype);
+                              snoe->snoe_evtype);
+            break;
         }
     }
+
+    /* Remove notify os event from pool */
+    os_memblock_put(&sensor_notify_evt_pool, snoe);
```

Comment:
Nitpick: I think this comment isn't quite right.  This line puts the event back into the pool rather than removing it.

Reply:
cool, will fix it.